cmd/root.go

```diff
@@ -66,34 +71,58 @@ var rootCmd = &cobra.Command{
 	},
 }
 
+// check if the version file exists in the container
+func checkVersionExists(versionUrl string) bool {
+	resp, err := http.Get(versionUrl)
+	if err != nil {
+		log.Err("checkVersionExists: error getting version file from container [%s]", err)
+		return false
+	}
+
+	return resp.StatusCode != 404
+}
+
+func getRemoteVersion(req string) (string, error) {
+	resp, err := http.Get(req)
+	if err != nil {
+		log.Err("getRemoteVersion: error listing version file from container [%s]", err)
+		return "", err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Err("getRemoteVersion: error reading body of response [%s]", err)
+		return "", err
+	}
+
+	var versionList ContainerList
+	err = xml.Unmarshal(body, &versionList)
+	if err != nil {
+		log.Err("getRemoteVersion: error unmarshalling xml response [%s]", err)
+		return "", err
+	}
+
+	if len(versionList.Blobs) != 1 {
+		return "", fmt.Errorf("the latest version container is empty")
```

Comment:
for size != 1 version container is empty is not a correct error message, we shall rather say not able to validate or some sort of generic error message.

Reply:
Done